.circleci/config.yml

```diff
@@ -28,12 +28,17 @@ workflows:
           #
           # The AWS ECR repo name, example: remind101/acme-inc
           repo: "$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME"
-          # tags for this image, comma separated.
-          tag: "latest,$CIRCLE_BRANCH,$CIRCLE_SHA1"
+          # tags for this image, comma separated. A composite of branch+sha1 is
+          # added so the lifecycle rule (below) can filter images beginning
+          # with "master" or "main" to avoid applying aggressive rules.
+          tag: "latest,$CIRCLE_BRANCH,$CIRCLE_SHA1,$CIRCLE_BRANCH-$CIRCLE_BUILD_NUM"
```

Comment:
I think I'd still want to keep the commit SHA in the prefixed tag, because it's nice to have that info embedded and also would make it easier to transition to something other than CircleCI without causing conflicts.